providers/fs/fs.go

```diff
@@ -0,0 +1,48 @@
+// Package fs implements a koanf.Provider that reads raw bytes
+// from given fs.FS to be used with a koanf.Parser to parse
+// into conf maps.
+
+// +build go1.16
+
+package fs
+
+import (
+	"errors"
+	"io/fs"
+	"io/ioutil"
+)
+
+// FS implements an fs.FS provider.
+type FS struct {
+	fs   fs.FS
+	path string
+}
+
+// Provider returns an fs.FS provider.
+func Provider(fs fs.FS, filepath string) *FS {
+	return &FS{fs: fs, path: filepath}
+}
+
+// ReadBytes reads the contents of given filepath from fs.FS and returns the bytes.
+func (f *FS) ReadBytes() ([]byte, error) {
+	fd, err := f.fs.Open(f.path)
+	if err != nil {
+		return nil, err
+	}
+	buf, err := ioutil.ReadAll(fd)
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close()
```

Comment:
`fd.Close()` belongs under L31?

Reply:
AFAIK `defer something.Close()` is typically called after error checking.

Reply:
I meant to say that it should be:

```go
	fd, err := f.fs.Open(f.path)
	if err != nil {
		return nil, err
	}
	defer fd.Close()

	buf, err := ioutil.ReadAll(fd)
	if err != nil {
		return nil, err
	}
```

instead of 

```go
	fd, err := f.fs.Open(f.path)
	if err != nil {
		return nil, err
	}
	buf, err := ioutil.ReadAll(fd)
	if err != nil {
		return nil, err
	}
	defer fd.Close()
```

Reply:
fixed